services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/client/grpc/DataPlaneClientImpl.java

```diff
@@ -16,47 +16,182 @@
 package com.futurewei.alcor.dataplane.client.grpc;
 
 import com.futurewei.alcor.dataplane.client.DataPlaneClient;
+import com.futurewei.alcor.dataplane.config.Config;
 import com.futurewei.alcor.dataplane.config.grpc.GoalStateProvisionerClient;
 import com.futurewei.alcor.dataplane.entity.HostGoalState;
+import com.futurewei.alcor.schema.GoalStateProvisionerGrpc;
 import com.futurewei.alcor.schema.Goalstate;
+import com.futurewei.alcor.schema.Goalstateprovisioner;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import io.netty.util.concurrent.DefaultThreadFactory;
+import org.apache.catalina.Manager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
-import java.util.List;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.stream.Collectors;
 
 @Component
 public class DataPlaneClientImpl implements DataPlaneClient {
 
+    private static final Logger LOG = LoggerFactory.getLogger(DataPlaneClientImpl.class);
+
     private int grpcPort;
 
     @Autowired
-    public DataPlaneClientImpl(@Value("${dataplane.grpc.port}")int grpcPort) {
-        this.grpcPort = grpcPort;
+    private Config globalConfig;
+
+    private final ExecutorService executor;
+
+    @Autowired
+    public DataPlaneClientImpl() {
+        this.grpcPort = globalConfig.port;
+        executor = new ThreadPoolExecutor(globalConfig.grpcMinThreads,
+                globalConfig.grpcMaxThreads, 50, TimeUnit.SECONDS,
+                new LinkedBlockingDeque<>(),
+                new DefaultThreadFactory(globalConfig.grpThreadsName));
```

Comment:
Looks like supporting thread pool is something missing in our Alcor async executor. Could you evaluate with @chenpiaoping to see what need to add into our own executor. 

Ideally we would like to replace it with our own implementation. I understand it may take some time but we will need to know what the gap is so that we could plan for it.